File: bmv-frontend/src/services/supabase.ts
// Service Supabase — client Auth + base de données
import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = import.meta.env.VITE_SUPABASE_URL as string;
const SUPABASE_ANON_KEY = import.meta.env.VITE_SUPABASE_ANON_KEY as string;

if (!SUPABASE_URL || !SUPABASE_ANON_KEY) {
  throw new Error(
    'Variables d\'environnement Supabase manquantes : VITE_SUPABASE_URL et VITE_SUPABASE_ANON_KEY sont requises.'
  );
}

// Client unique partagé dans toute l'app
export const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY);

/**
 * Connexion Google OAuth via Supabase Auth.
 * redirectTo utilise window.location.origin pour fonctionner depuis n'importe quelle
 * adresse (localhost en dev PC, IP réseau depuis un téléphone, domaine en production).
 * skipBrowserRedirect: false force Supabase à suivre le redirectTo fourni
 * plutôt que de tomber en fallback sur le Site URL configuré dans le dashboard.
 */
export const signInWithGoogle = async () => {
  const { error } = await supabase.auth.signInWithOAuth({
    provider: 'google',
    options: {
      redirectTo: `${window.location.origin}/auth/callback`,
      skipBrowserRedirect: false,
    },
  });
  if (error) throw error;
};

/**
 * Déconnexion
 */
export const signOut = async () => {
  const { error } = await supabase.auth.signOut();
  if (error) throw error;
};

File: bmv-backend/app/services/supabase.py
# Service Supabase — client backend avec la service key
# La service key bypass le Row Level Security : à utiliser uniquement côté serveur

from supabase import create_client, Client
from app.config import settings

def get_supabase_client() -> Client:
    """Retourne un client Supabase authentifié avec la service key."""
    if not settings.supabase_url or not settings.supabase_service_key:
        raise ValueError(
            "Variables d'environnement Supabase manquantes : "
            "SUPABASE_URL et SUPABASE_SERVICE_KEY sont requises."
        )
    return create_client(settings.supabase_url, settings.supabase_service_key)

# Instance unique partagée dans toute l'app
supabase: Client = get_supabase_client()
